c_api/impl/AuxIndexStructures_c.cpp

```diff
@@ -27,6 +28,9 @@ using faiss::IDSelectorXOr;
 using faiss::RangeQueryResult;
 using faiss::RangeSearchPartialResult;
 using faiss::RangeSearchResult;
+using faiss::IndexIDMap;
+using faiss::IndexIDMap2;
```

Comment:
cherry picked from main to avoid merge conflicts - 
earlier I had made some changes to the selector(IDMap -> IDMap2) and copied the code. Now, I've cherrypicked it without that to avoid merge conflicts.